enhancements/microshift/microshift-pluggable-core-components.md

```diff
@@ -0,0 +1,213 @@
+---
+title: microshift-pluggable-core-components
+authors:
+  - majopela
+reviewers:
+  - "@copejon, MicroShift contributor"
+  - "@fzdarsky, MicroShift architect"
+  - "@ggiguash, MicroShift contributor"
+  - "@dhellmann, MicroShift contributor"
+  - "@oglok, MicroShift contributor"
+  - "@zshi-redhat, MicroShift "
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2022-12-19
+last-updated: 2022-12-19
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-599
+---
+
+# MicroShift Pluggable Core Components
+
+## Summary
+
+This enhancement proposes making some of the core components like CNI, CSI or router
+optional and pluggable.
+
+In the enhancement proposal we will describe an architecture which would allow
+customers to install individual compomnents as rpm packages.
+
+
+## Motivation
+
+The principal factor that gave birth to the idea of MicroShift was the need to bring
+the OpenShift deployment model into the Edge where the power, memory and CPU budgets
+are reduced.
```

Comment:
Another motivation is to upstream MicroShift where we would require different CSI and CNI drivers.
Can we add this factor as well?